File: app/src/main/java/com/bq/zowi/interactors/ForgetZowiInteractorImpl.java
package com.bq.zowi.interactors;

import com.bq.zowi.controllers.BTConnectionController;
import com.bq.zowi.controllers.SessionController;
import rx.Single;
import rx.functions.Func1;

/* JADX INFO: loaded from: classes.dex */
public class ForgetZowiInteractorImpl implements ForgetZowiInteractor {
    private final ChangeZowiNameInteractor changeZowiNameInteractor;
    private final BTConnectionController connectionController;
    private final SessionController sessionController;

    public ForgetZowiInteractorImpl(ChangeZowiNameInteractor changeZowiNameInteractor, SessionController sessionController, BTConnectionController connectionController) {
        this.changeZowiNameInteractor = changeZowiNameInteractor;
        this.sessionController = sessionController;
        this.connectionController = connectionController;
    }

    @Override // com.bq.zowi.interactors.ForgetZowiInteractor
    public Single<Void> forgetZowi() {
        return this.changeZowiNameInteractor.resetZowiNameToFactory().map(new Func1<Void, Void>() { // from class: com.bq.zowi.interactors.ForgetZowiInteractorImpl.1
            @Override // rx.functions.Func1
            public Void call(Void aVoid) {
                ForgetZowiInteractorImpl.this.sessionController.resetActiveZowi();
                ForgetZowiInteractorImpl.this.connectionController.stopConnection();
                return null;
            }
        });
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/usecases/ForgetZowiInteractorImpl.kt
package com.bq.zowi.usecases

import com.bq.zowi.api.BTConnectionController
import com.bq.zowi.api.SessionController
import io.reactivex.Completable

class ForgetZowiInteractorImpl(
    private val changeZowiNameInteractor: ChangeZowiNameInteractor,
    private val sessionController: SessionController,
    private val connectionController: BTConnectionController
) : ForgetZowiInteractor {

    override fun forgetZowi(): Completable {
        return changeZowiNameInteractor.resetZowiNameToFactory()
            .andThen(Completable.fromAction {
                sessionController.resetActiveZowi()
                connectionController.stopConnection()
            })
    }
}
